lib/logger/test/logger_test.exs

```diff
@@ -122,6 +122,14 @@ defmodule LoggerTest do
     end) =~ msg("application= module=LoggerTest [info]  ok")
   end
 
+  test "metadata merge with function returned metadata" do
```

Comment:
This test name looks a bit confusing to me. Can we call it something like "metadata merge when the argument function returns metadata"?